### PR TITLE
fix: ever increasing idle http connections while fetching transformer features

### PR DESCRIPTION
**Pull request by cisse21**
# Description

Use a single http client to make transformer fetch call
Not using the default transport provided by Go 
Renaming few fields in the features implementation 

## Linear Ticket

Fixes PIPE-1019

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: services/transformer/features.go [hunk 1]
```diff
@@ -38,12 +40,21 @@ var defaultTransformerFeatures = `{
...
+func NewFeaturesService(ctx context.Context, config *config.Config, featConfig FeaturesServiceConfig) FeaturesService {
 	handler := &featuresService{
 		features: json.RawMessage(defaultTransformerFeatures),
 		logger:   logger.NewLogger().Child("transformer-features"),
 		waitChan: make(chan struct{}),
-		config:   config,
+		options:  featConfig,
+		client: &http.Client{
+			Transport: &http.Transport{
+				DisableKeepAlives:   config.GetBool("Transformer.Client.disableKeepAlives", true),
```
Comment (lvrach) [inline services/transformer/features.go:51]: shouldn't this be false by default?
```suggestion
				DisableKeepAlives:   config.GetBool("Transformer.Client.disableKeepAlives", false),
```

**Review by lvrach — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

File: services/transformer/features_impl.go [hunk 2]
```diff
@@ -11,16 +11,16 @@ import (
...
-	"github.com/rudderlabs/rudder-go-kit/config"
 	"github.com/rudderlabs/rudder-go-kit/logger"
 	"github.com/rudderlabs/rudder-server/utils/httputil"
 )
 
 type featuresService struct {
 	logger   logger.Logger
 	waitChan chan struct{}
-	config   FeaturesServiceConfig
+	options  FeaturesServiceConfig
```
Comment (mihir20) [inline services/transformer/features_impl.go:21]: Should we rename this struct as well?
```suggestion
	options  FeaturesServiceOptions
```

**Review by mihir20 — APPROVED** [review]

**Review by achettyiitr — APPROVED** [review]

File: services/transformer/features.go (hunk 1 above)
Comment (cisse21) [reply services/transformer/features.go:51]: I see it to be true on transformer [here](https://github.com/rudderlabs/rudder-server/blob/master/processor/transformer/transformer.go#L221)

File: services/transformer/features_impl.go (hunk 2 above)
Comment (cisse21) [reply services/transformer/features_impl.go]: Done

File: services/transformer/features.go (hunk 1 above)
Comment (cisse21) [reply services/transformer/features.go]: Changed it to false anyways

**Pull request merged**